language/analyze.js

```diff
@@ -1,304 +1,220 @@
+    // The document text, e.g. "Hello, world!"
+    content: text
   });
-}
-// [END analyze_syntax_from_string]
 
-// [START analyze_syntax_from_file]
-/**
- * Detect the syntax in a text file that resides in Google Cloud Storage.
- *
- * @param {string} bucket The bucket where the file resides.
- * @param {string} filename The name of the file to be analyzed.
- * @param {object} [options] Optional configuration.
- * @param {string} [options.language] The language of the text, e.g. "en".
- * @param {string} [options.type] The type of the text, either "text" or "html".
- * @param {function} callback The callback function.
- */
-function analyzeSyntaxFromFile (bucket, filename, options, callback) {
-  var document = language.document({
-    content: storage.bucket(bucket).file(filename),
-    type: options.type,
-    language: options.language
+  // Detects syntax in the document
+  return document.detectSyntax()
+    .then((results) => {
+      const syntax = results[0];
+      console.log('Tags:');
+      syntax.forEach((part) => console.log(part.tag));
+      return syntax;
+    });
+}
+// [END language_syntax_string]
+
+// [START language_syntax_file]
+function analyzeSyntaxInFile (bucketName, fileName) {
+  // Instantiates clients
+  const language = Language();
+  const storage = Storage();
+
+  // The bucket where the file resides, e.g. "my-bucket"
+  const bucket = storage.bucket(bucketName);
+  // The text file to analyze, e.g. "file.txt"
+  const file = bucket.file(fileName);
+
+  // Instantiates a Document, representing a text file in Cloud Storage
+  const document = language.document({
+    // The GCS file
+    content: file
   });
 
-  var config = {
-    syntax: true
-  };
-
-  // See https://googlecloudplatform.github.io/google-cloud-node/#/docs/language/latest/language/document
-  document.annotate(config, function (err, result, apiResponse) {
-    if (err) {
-      return callback(err);
-    }
-
-    console.log('Done analyzing syntax');
-    return callback(null, apiResponse);
-  });
+  // Detects syntax in the document
+  return document.detectSyntax()
+    .then((results) => {
+      const syntax = results[0];
+      console.log('Tags:');
+      syntax.forEach((part) => console.log(part.tag));
+      return syntax;
+    });
 }
-// [END analyze_syntax_from_file]
-// [END all]
+// [END language_syntax_file]
 
 // The command-line program
-var cli = require('yargs');
-var utils = require('../utils');
-
-var program = module.exports = {
-  analyzeSentimentFromString: analyzeSentimentFromString,
-  analyzeSentimentFromFile: analyzeSentimentFromFile,
-  analyzeEntitiesFromString: analyzeEntitiesFromString,
-  analyzeEntitiesFromFile: analyzeEntitiesFromFile,
-  analyzeSyntaxFromString: analyzeSyntaxFromString,
-  analyzeSyntaxFromFile: analyzeSyntaxFromFile,
-  main: function (args) {
+const cli = require(`yargs`);
+
+const program = module.exports = {
+  analyzeSentimentOfText: analyzeSentimentOfText,
```

Comment:
With ES6, do we need to do `analyze...Text: analyze...Text`? (Can't we just do `analyze...Text`?)